libraries/AP_Scheduler/PerfInfo.h

```diff
@@ -26,6 +36,25 @@ class PerfInfo {
 
     void update_logging();
 
+    // allocate the task info array
+    void allocate_task_info(uint8_t num_tasks);
+    // free the task info array
+    void free_task_info();
+    // whether or not we have task info
+    bool has_task_info() { return _task_info != nullptr; }
+    // return a task info
+    const TaskInfo* get_task_info(uint8_t task_index) const {
+        return (_task_info && task_index < _num_tasks) ? &_task_info[task_index] : nullptr;
+    }
+    // update task info
```

Comment:
Yep, would really have never guessed that's what `update_task_info` does :-P

Please don't add comments that restate the C++ in English.

```suggestion
    // update_task_info - called after each run of a task to update its statistics based on measurements taken by the scheduler
```

Similarly for other comments here.

Reply:
Fixed